ggml/src/ggml-cuda/mmf.cu

```diff
@@ -65,15 +100,38 @@ static __global__ void mul_mat_f(
                 for (int j0 = 0; j0 < tile_B::I; ++j0) {
                     const int j = j0 + itB*tile_B::I;
 
-                    tile_xy[j0*tile_k_padded + threadIdx.x] = j < cols_per_block ? y[j*stride_col_y + col] : 0.0f;
+                    if (!has_ids) {
```

Comment:
Without having run the code myself I'm pretty certain that this conditional statement will impact the performance of dense models. It would probably make sense to make `has_ids` a template parameter.